File: src/main/java/com/khalil/saidane/ecommerce/entities/Supplier.java
package com.khalil.saidane.ecommerce.entities;

import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
public class Supplier {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String contactName;
    private  String city;
    private String state;
    private String zipCode;
    private String country;
    private int phone;
    private String Email;
    private String paymentMethods;
    private String discountType;
    private String typeGoods;
    @OneToMany(mappedBy = "supplier")
    private List<Product> products;



}

File: E-Commerce-Front/e-commerce-front/src/app/module/Supplier.ts
import {Product} from './Product';

export interface Supplier {
  id: number;
  name: string;
  contactName: string;
  city: string;
  state: string;
  zipCode: string;
  country: string;
  phone: number;
  Email: string;
  paymentMethods: string;
  discountType: string;
  typeGoods: string;
  products: Product[];
}
